File: src/main/java/eu/mshade/enderframe/item/metadata/NameItemStackMetadata.java
package eu.mshade.enderframe.item.metadata;

import eu.mshade.enderframe.metadata.MetadataKeyValue;
import eu.mshade.enderframe.metadata.itemstack.ItemStackMetadataKey;
import eu.mshade.enderframe.mojang.chat.TextComponent;

public class NameItemStackMetadata extends MetadataKeyValue<TextComponent> {

    public NameItemStackMetadata(TextComponent metadataValue) {
        super(ItemStackMetadataKey.NAME, metadataValue);
    }

    public NameItemStackMetadata(String name) {
        this(TextComponent.of(name));
    }

}

File: src/main/java/eu/mshade/enderframe/item/metadata/NameItemStackMetadata.kt
package eu.mshade.enderframe.item.metadata

import eu.mshade.enderframe.item.ItemStackMetadataKey
import eu.mshade.enderframe.metadata.MetadataKeyValue
import eu.mshade.enderframe.mojang.chat.TextComponent

class NameItemStackMetadata(metadataValue: TextComponent) :
    MetadataKeyValue<TextComponent?>(ItemStackMetadataKey.NAME, metadataValue) {

    constructor(name: String) : this(TextComponent.of(name))

}
